Static lifetime is difficult to work with

Consider some code like:

```rust
diesel::update(&alert)
    .set((active.eq(Option::<bool>::None), end.eq(diesel::dsl::now)))
    .execute_async(&pool)
    .await;
```

where `alert` is a reference to a struct that implements `Identifiable`.

This code fails with an error like:

```
error[E0597]: `alert` does not live long enough
  --> iml-services/iml-ntp/src/main.rs:47:36
   |
47 |                     diesel::update(&alert)
   |                     ---------------^^^^^^-
   |                     |              |
   |                     |              borrowed value does not live long enough
   |                     argument requires that `alert` is borrowed for `'static`
...
51 |                 }
   |                 - `alert` dropped here while still borrowed
```

I may be missing something obvious but it doesn't seem like it's possible to use non-static references when building a query with this lib.

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -9,13 +9,14 @@ license = "MIT/Apache-2.0"
 categories = ["asynchronous", "database"]
 
 [dependencies]
-async-trait = "0.1.21"
-diesel = { version = "1.4.3", features = [ "r2d2" ] }
-futures = "0.3.1"
-r2d2 = "0.8.7"
-tokio = { version = "0.2.4", features = [ "blocking" ] }
+async-trait = "0.1.30"
+diesel = { version = "1.4.4", features = [ "r2d2" ] }
+futures = "0.3.4"
+r2d2 = "0.8.8"
+tokio = { version = "0.2.17", features = [ "rt-threaded", "blocking" ] }
 
 [dev-dependencies]
-diesel = { version = "1.4.3", features = [ "postgres", "uuidv07" ] }
-uuid = { version = "0.7.4", features = [ "v4" ] }
-tokio = { version = "0.2.4", default-features = false, features = [ "full" ] }
+diesel = { version = "1.4.4", features = [ "postgres", "uuidv07" ] }
+uuid = { version = "0.8.1", features = [ "v4" ] }
+tokio = { version = "0.2.17", default-features = false, features = [ "full" ] }
+

--- a/src/lib.rs
+++ b/src/lib.rs
@@ -73,12 +73,10 @@ where
     async fn batch_execute_async(&self, query: &str) -> AsyncResult<()> {
         let self_ = self.clone();
         let query = query.to_string();
-        task::spawn_blocking(move || {
+        task::block_in_place(move || {
             let conn = self_.get().map_err(AsyncError::Checkout)?;
             conn.batch_execute(&query).map_err(AsyncError::Error)
         })
-        .await
-        .expect("task has panicked")
     }
 }
 
@@ -89,13 +87,13 @@ where
 {
     async fn run<R, Func>(&self, f: Func) -> AsyncResult<R>
     where
-        R: 'static + Send,
-        Func: 'static + FnOnce(&Conn) -> QueryResult<R> + Send;
+        R: Send,
+        Func: FnOnce(&Conn) -> QueryResult<R> + Send;
 
     async fn transaction<R, Func>(&self, f: Func) -> AsyncResult<R>
     where
-        R: 'static + Send,
-        Func: 'static + FnOnce(&Conn) -> QueryResult<R> + Send;
+        R: Send,
+        Func: FnOnce(&Conn) -> QueryResult<R> + Send;
 }
 
 #[async_trait]
@@ -106,31 +104,27 @@ where
     #[inline]
     async fn run<R, Func>(&self, f: Func) -> AsyncResult<R>
     where
-        R: 'static + Send,
-        Func: 'static + FnOnce(&Conn) -> QueryResult<R> + Send,
+        R: Send,
+        Func: FnOnce(&Conn) -> QueryResult<R> + Send,
     {
         let self_ = self.clone();
-        task::spawn_blocking(move || {
+        task::block_in_place(move || {
             let conn = self_.get().map_err(AsyncError::Checkout)?;
             f(&*conn).map_err(AsyncError::Error)
         })
-        .await
-        .expect("task has panicked")
     }
 
     #[inline]
     async fn transaction<R, Func>(&self, f: Func) -> AsyncResult<R>
     where
-        R: 'static + Send,
-        Func: 'static + FnOnce(&Conn) -> QueryResult<R> + Send,
+        R: Send,
+        Func: FnOnce(&Conn) -> QueryResult<R> + Send,
     {
         let self_ = self.clone();
-        task::spawn_blocking(move || {
+        task::block_in_place(move || {
             let conn = self_.get().map_err(AsyncError::Checkout)?;
             conn.transaction(|| f(&*conn)).map_err(AsyncError::Error)
         })
-        .await
-        .expect("task has panicked")
     }
 }
 
@@ -145,22 +139,22 @@ where
 
     async fn load_async<U>(self, asc: &AsyncConn) -> AsyncResult<Vec<U>>
     where
-        U: 'static + Send,
+        U: Send,
         Self: LoadQuery<Conn, U>;
 
     async fn get_result_async<U>(self, asc: &AsyncConn) -> AsyncResult<U>
     where
-        U: 'static + Send,
+        U: Send,
         Self: LoadQuery<Conn, U>;
 
     async fn get_results_async<U>(self, asc: &AsyncConn) -> AsyncResult<Vec<U>>
     where
-        U: 'static + Send,
+        U: Send,
         Self: LoadQuery<Conn, U>;
 
     async fn first_async<U>(self, asc: &AsyncConn) -> AsyncResult<U>
     where
-        U: 'static + Send,
+        U: Send,
         Self: LimitDsl,
         Limit<Self>: LoadQuery<Conn, U>;
 }
@@ -168,7 +162,7 @@ where
 #[async_trait]
 impl<T, Conn> AsyncRunQueryDsl<Conn, Pool<ConnectionManager<Conn>>> for T
 where
-    T: 'static + Send + RunQueryDsl<Conn>,
+    T: Send + RunQueryDsl<Conn>,
     Conn: 'static + Connection,
 {
     async fn execute_async(self, asc: &Pool<ConnectionManager<Conn>>) -> AsyncResult<usize>
@@ -180,7 +174,7 @@ where
 
     async fn load_async<U>(self, asc: &Pool<ConnectionManager<Conn>>) -> AsyncResult<Vec<U>>
     where
-        U: 'static + Send,
+        U: Send,
         Self: LoadQuery<Conn, U>,
     {
         asc.run(|conn| self.load(&*conn)).await
@@ -188,7 +182,7 @@ where
 
     async fn get_result_async<U>(self, asc: &Pool<ConnectionManager<Conn>>) -> AsyncResult<U>
     where
-        U: 'static + Send,
+        U: Send,
         Self: LoadQuery<Conn, U>,
     {
         asc.run(|conn| self.get_result(&*conn)).await
@@ -196,7 +190,7 @@ where
 
     async fn get_results_async<U>(self, asc: &Pool<ConnectionManager<Conn>>) -> AsyncResult<Vec<U>>
     where
-        U: 'static + Send,
+        U: Send,
         Self: LoadQuery<Conn, U>,
     {
         asc.run(|conn| self.get_results(&*conn)).await
@@ -204,7 +198,7 @@ where
 
     async fn first_async<U>(self, asc: &Pool<ConnectionManager<Conn>>) -> AsyncResult<U>
     where
-        U: 'static + Send,
+        U: Send,
         Self: LimitDsl,
         Limit<Self>: LoadQuery<Conn, U>,
     {

